data-model/src/test/scala/za/co/absa/enceladus/model/menas/audit/AuditableTest.scala

```diff
@@ -75,4 +84,21 @@ class AuditableTest extends AnyFunSuite {
           AuditTrailChange(field = "conformance", oldValue = Some("LiteralConformanceRule(0,something,true,1.01)"), None, message = "Conformance rule removed.")
     ))(removedRes)
   }
+
+  test("Testing added getOptionalMapFieldsAudit") {
+    val result = obj2.getOptionalMapFieldsAudit(obj3, AuditFieldName("properties", "Property"))
+    assertResult(Seq(
+      AuditTrailChange(field = "properties", newValue = Some("""key: "key1" value: "value1""""), oldValue = None, message = "Property added.")
+    ))(result)
+  }
+
+  test("Testing changed getOptionalMapFieldsAudit") {
+    val newObj2 = obj2.copy(properties = Some(Map("key1" -> "value previous")))
+    val result = newObj2.getOptionalMapFieldsAudit(obj3, AuditFieldName("properties", "Property"))
+    assertResult(Seq(
+      AuditTrailChange(field = "properties", newValue = Some("""key: "key1" value: "value1""""), oldValue = Some("""key: "key1" value: "value previous""""), message = "Property changed.")
+    ))(result)
+  }
+
+
```

Comment:
I believe the _removed_ test case should also be present.

```suggestion
  test("Testing removed getOptionalMapFieldsAudit") {
    val result = obj3.getOptionalMapFieldsAudit(obj2, AuditFieldName("properties", "Property"))
    assertResult(Seq(
      AuditTrailChange(field = "properties", newValue = None, oldValue = Some("""key: "key1" value: "value1""""), message = "Property removed.")
    ))(result)
  }
```

Reply:
Depending on the rest of the decision, you may want to include the cases for both `properties = Some(Map.empty)|None` for the _added_ and _removed_ test cases, to make sure that you correctly cover the trail creation for these cases, too.

Whether to spell it out and forego some DRYness or use the infamous `Seq(None, Some(Map.empty)).forEach{ props => test(s"almost the same test differentiated by $prop")}` is up to you.

For my brief'n'naive™ testing I have just spelled it out (steal away if it seems useful):
```scala
  test("Testing added getOptionalMapFieldsAudit - explicitly None properties") {
    val result = obj2.copy(properties = None).getOptionalMapFieldsAudit(obj3, AuditFieldName("properties", "Property"))
    assertResult(Seq(
      AuditTrailChange(field = "properties", newValue = Some("""key: "key1" value: "value1""""), oldValue = None, message = "Property added.")
    ))(result)
  }

  test("Testing removed getOptionalMapFieldsAudit  - explicitly None properties") {
    val result = obj3.getOptionalMapFieldsAudit(obj2.copy(properties = None), AuditFieldName("properties", "Property"))
    assertResult(Seq(
      AuditTrailChange(field = "properties", newValue = None, oldValue = Some("""key: "key1" value: "value1""""), message = "Property removed.")
    ))(result)
  }
```

Reply:
A great catch. Did not notice properties were Empty by default.